packages/node_modules/@ciscospark/internal-plugin-reaction/src/reaction.js

```diff
@@ -0,0 +1,125 @@
+/*!
+ * Copyright (c) 2015-2017 Cisco Systems, Inc. See LICENSE file.
+ */
+
+import {SparkPlugin} from '@ciscospark/spark-core';
+import uuid from 'uuid';
+
+/**
+ * @typedef {Object} ReactionObject
+ * @property {object} participant - an actor with an id and entryUUID
+ * @property {string} activityID - the id of the activity being reacted upon
+ * @property {string} convoID - the id of the target conversation
+ * @property {string} targetURL - url of the target conversation
+ * @property {string} reactionID - (optional) the id of the reaction to be changed
+ */
+
+const Reaction = SparkPlugin.extend({
+  namespace: 'Reaction',
+
+  /**
+  * helper function to create the request object
+  * @param {Object} options
+  * @returns {Object} the reaction request object
+  */
+  createReactionRequestBody(options) {
+    return {
+      clientTempId: uuid.v4(),
+      objectType: 'activity',
+      verb: options.verb,
+      actor: {
+        objectType: 'person',
+        id: options.participant.id,
+        entryUUID: options.participant.entryUUID
+      },
+      object: {
+        objectType: 'messageProperty',
+        tags: ['REACTIONS'],
+        id: options.activityID,
+        reactionType: options.reactionID
+      },
+      target: {
+        id: options.convoID,
+        url: options.targetURL,
+        objectType: 'conversation'
+      }
+    };
+  },
+
+  /**
+  * helper function to validate options parameter
+  * @param {Object} options
+  * @returns {string} the rejection error message or null if options is good
+  */
+  validateReaction(options) {
```

Comment:
Still would like the options object to be a `@typedef` defined jsdoc object, something like:
```js
@typedef {Object} ReactionOptionsObject
@property {string} activityId
@property {string} convoId
```

There are quite a few throughout the sdk that we try to model off of.